### PR TITLE
feature/OAuth2

### PR DESCRIPTION
**Pull request by a3828162**
- [x] Use Middleware for authorization
- [x] Consumer retrieve token
- [x] Replace the service name in each HTTP request of the producer with the OpenAPI name
- [x] Complete unit test of router_auth_check_test.go

### CODE REVIEW

File: internal/sbi/producer/generate_auth_data.go [hunk 1]
```diff
@@ -461,8 +469,10 @@ func GenerateAuthDataProcedure(authInfoRequest models.AuthenticationInfoRequest,
 	}
 
 	var rsp *http.Response
+	// rsp, err = client.AuthenticationDataDocumentApi.ModifyAuthentication(
```
Comment (ianchen0119) [inline internal/sbi/producer/generate_auth_data.go:472]: please remove commented code

File: internal/sbi/producer/parameter_provision.go [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 	"net/http"
 
 	"github.com/free5gc/openapi"
 	"github.com/free5gc/openapi/models"
+	udm_context "github.com/free5gc/udm/internal/context"
 	"github.com/free5gc/udm/internal/logger"
 	"github.com/free5gc/util/httpwrapper"
 )
 
+// const serviceName string = string(models.ServiceName_NUDR_DR)
```
Comment (ianchen0119) [inline internal/sbi/producer/parameter_provision.go:13]: please remove commented code

File: internal/sbi/producer/subscriber_data_management.go [hunk 3]
```diff
@@ -295,7 +308,9 @@ func getSupiProcedure(supi string, plmnID string, dataSetNames []string, support
 	if containDataSetName(dataSetNames, string(models.DataSetName_SMF_SEL)) {
 		var smfSelSubsbody models.SmfSelectionSubscriptionData
 		udm_context.GetSelf().CreateSmfSelectionSubsDataforUe(supi, smfSelSubsbody)
-		smfSelData, res, err := clientAPI.SMFSelectionSubscriptionDataDocumentApi.QuerySmfSelectData(context.Background(),
+		// smfSelData, res, err := clientAPI.SMFSelectionSubscriptionDataDocumentApi.QuerySmfSelectData(context.Background(),
```
Comment (ianchen0119) [inline internal/sbi/producer/subscriber_data_management.go:311]: please remove commented code

File: internal/sbi/producer/subscriber_data_management.go [hunk 4]
```diff
@@ -339,8 +354,10 @@ func getSupiProcedure(supi string, plmnID string, dataSetNames []string, support
 		var querySmfRegListParamOpts Nudr.QuerySmfRegListParamOpts
 		querySmfRegListParamOpts.SupportedFeatures = optional.NewString(supportedFeatures)
 		udm_context.GetSelf().CreateUeContextInSmfDataforUe(supi, UeContextInSmfbody)
+		// pdusess, res, err := clientAPI.SMFRegistrationsCollectionApi.QuerySmfRegList(
```
Comment (ianchen0119) [inline internal/sbi/producer/subscriber_data_management.go:357]: please remove commented code

File: internal/sbi/producer/subscriber_data_management.go [hunk 5]
```diff
@@ -403,8 +420,10 @@ func getSupiProcedure(supi string, plmnID string, dataSetNames []string, support
 	// }
 
 	if containDataSetName(dataSetNames, string(models.DataSetName_SM)) {
+		// sessionManagementSubscriptionData, res, err := clientAPI.SessionManagementSubscriptionDataApi.
```
Comment (ianchen0119) [inline internal/sbi/producer/subscriber_data_management.go:423]: please remove commented code

File: internal/sbi/producer/subscriber_data_management.go [hunk 6]
```diff
@@ -446,8 +465,10 @@ func getSupiProcedure(supi string, plmnID string, dataSetNames []string, support
 	if containDataSetName(dataSetNames, string(models.DataSetName_TRACE)) {
 		var TraceDatabody models.TraceData
 		udm_context.GetSelf().CreateTraceDataforUe(supi, TraceDatabody)
+		// traceData, res, err := clientAPI.TraceDataDocumentApi.QueryTraceData(
```
Comment (ianchen0119) [inline internal/sbi/producer/subscriber_data_management.go:468]: please remove commented code

**Review by ianchen0119 — CHANGES_REQUESTED** [review]
Hi @a3828162 
Please update the patch base on my comments.
Thanks.

**Review by andy89923 — APPROVED** [review]
LGTM!

File: internal/util/util.go [hunk 7]
```diff
@@ -13,6 +13,23 @@ const (
...
+	ServiceNameNudmEe   = string(models.ServiceName_NUDM_EE)
+	ServiceNameNudmPp   = string(models.ServiceName_NUDM_PP)
+	ServiceNameNudmSdm  = string(models.ServiceName_NUDM_SDM)
+	ServiceNameNudmUeau = string(models.ServiceName_NUDM_UEAU)
+	ServiceNameNudmUecm = string(models.ServiceName_NUDM_UECM)
+	ServiceNameNudrDr   = string(models.ServiceName_NUDR_DR)
+)
+
+const (
+	NfTypeNRF = models.NfType_NRF
```
Comment (tim-ywliu) [inline internal/util/util.go:28]: no need to define const again, directly use models

File: internal/util/util.go [hunk 8]
```diff
@@ -13,6 +13,23 @@ const (
 	UdmDefaultConfigPath = "./config/udmcfg.yaml"
 )
 
+const (
+	ServiceNameNnrfDisc = string(models.ServiceName_NNRF_DISC)
```
Comment (tim-ywliu) [inline internal/util/util.go:17]: no need to define const again, directly use `string(models.ServiceNAme_xxx)`

**Review by ianchen0119 — APPROVED** [review]

**Pull request merged**